### DISCUSSION TITLE
Enabling query string for table filtering

### DISCUSSION THREAD

**Discussion by snarexx** [filamentphp/filament#10621] [Help]
I want the URL to include a query string when I filter a table to share it with others easily. I watched a video that has been shared that demonstrated this feature, and I was wondering how I can enable it. https://github.com/filamentphp/filament/issues/7129#issuecomment-1661456931 I use table builder on a public page, not table builder inside a panel

**Comment by snarexx** [comment]
This is the solution
https://filamentphp.com/docs/3.x/tables/advanced#query-string

**Reply by damms005** [reply]
@danharrin this doesn't work in resource pages list view. Am I doing something wrong or currently not supported, please?

**Reply by danharrin** [reply]
The filters are already in the query string by default for list views

**Reply by damms005** [reply]
I get below on latest version, notice how I filtered the `Transaction reference` column with string `abc` but the url still only has the active tab filter, not the column filter:

![image](https://github.com/user-attachments/assets/28c3d4dc-57bf-48de-b5f0-a85e6c45d3bb)

**Reply by danharrin** [reply]
Please open an issue with minimal reproduction repository

**Reply by damms005** [reply]
[Done](https://github.com/filamentphp/filament/issues/15660)

**Reply by KurlenkovEvgeniy** [reply]
If someone got here, you should add in your ListRecords this:
```PHP
use Livewire\Attributes\Url;

#[Url]
public array $tableColumnSearches = [];
```

**Reply by jalexmelendez** [reply]
> If someone got here, you should add in your ListRecords this:
> 
> ```
> use Livewire\Attributes\Url;
> 
> #[Url]
> public array $tableColumnSearches = [];
> ```

This works! (tested on filament v4) 

Thank you so much!
